server/gc/src/main/java/org/apache/accumulo/gc/GCRun.java

```diff
@@ -160,6 +160,16 @@ public Stream<Reference> getReferences() {
 
     // there is a lot going on in this "one line" so see below for more info
     var tabletReferences = tabletStream.flatMap(tm -> {
+
+      // verify that dir and prev row entries present for to check for complete row scan
+      log.trace("tablet metadata table id: {}, dir:{}, saw: {}, prev row: {}", tm.getTableId(),
+          tm.getDirName(), tm.sawPrevEndRow(), tm.getPrevEndRow());
+      if (tm.getDirName() == null || tm.getDirName().isEmpty() || !tm.sawPrevEndRow()) {
+        throw new IllegalStateException(
+            "possible incomplete metadata scan for table id: " + tm.getTableId() + ", dir: "
```

Comment:
For informational purposes it would be good to include the endRow.  That info comes from the row in the metadata table so should always be there.

```suggestion
            "possible incomplete metadata scan for table id: " + tm.getTableId() +" end row:"+tm.getEndRow()+", dir: "
```

Reply:
Added in 0d6f821680.  (Also included in trace log message.)